File: lambda/todo-read.js
'use strict';

const AWS = require('aws-sdk');
const dynamo = new AWS.DynamoDB.DocumentClient();

const tableName = process.env.TABLE_NAME;

const createResponse = (statusCode, body) => {
    return {
        statusCode: statusCode,
        body: JSON.stringify(body)
    }
};

exports.lambda_handler = (event, context, callback) => {

    let params = {};
    if (event.pathParameters.resourceId !== null) {
        params = {
            TableName: tableName,
            Key: {
                todo_id: event.pathParameters.resourceId
            }
        };
    } else {
        params = {
            TableName: tableName
        };
    }

    let dbGet = (params) => {
        return dynamo.get(params).promise()
    };

    dbGet(params).then((data) => {
        if (!data.Item) {
            callback(null, createResponse(404, "ITEM NOT FOUND"));
            return;
        }
        console.log(`RETRIEVED ITEM SUCCESSFULLY WITH doc = ${data.Item}`);
        callback(null, createResponse(200, data.Item));
    }).catch((err) => {
        console.log(`GET ITEM FAILED FOR doc = ${params.Key.id}, WITH ERROR: ${err}`);
        callback(null, createResponse(500, err));
    });
};

File: lambda/todo-read.py
from __future__ import print_function  # Python 2/3 compatibility
import boto3
from botocore.exceptions import ClientError
import json
import logging
import os

logger = logging.getLogger()
logger.setLevel(logging.INFO)


def lambda_handler(event, context):
    if 'resourceId' in event:
        return getItem(event['pathParameters']['resourceId'])
    else:
        return scan()


def getItem(id):
    try:
        table_name = os.environ['TABLE_NAME']
        dynamodb = boto3.resource('dynamodb')
        table = dynamodb.Table(table_name)
        result_todo = table.get_item(
            Key={
                'todo_id': id
            }
        )
        response = {
            "statusCode": "200",
            "headers": {
                'Access-Control-Allow-Origin': os.environ['CORS'],
                'Content-Type': 'application/json'
            },
            "body": json.dumps(result_todo["Item"])
        }
        return response
    except ClientError as e:
        logger.error(e.response['Error']['Message'])
        response = {
            "statusCode": "400",
            "headers": {
                'Access-Control-Allow-Origin': os.environ['CORS'],
                'Content-Type': 'application/json'
            },
            "message": json.dumps(e.response['Error']['Message'])
        }
        return response


def scan():
    try:
        table_name = os.environ['TABLE_NAME']
        dynamodb = boto3.resource('dynamodb')
        table = dynamodb.Table(table_name)
        items = table.scan()
        response = {
            "statusCode": "200",
            "headers": {
                'Access-Control-Allow-Origin': os.environ['CORS'],
                'Content-Type': 'application/json'
            },
            "body": json.dumps(items["Items"])
        }
        return response
    except ClientError as e:
        logger.error(e.response['Error']['Message'])
        response = {
            "statusCode": "400",
            "headers": {
                'Access-Control-Allow-Origin': os.environ['CORS'],
                'Content-Type': 'application/json'
            },
            "message": json.dumps(e.response['Error']['Message'])
        }
        return response
